src/dcurl.c

```diff
@@ -60,26 +70,44 @@ static uv_sem_t notify_remote;
 
 bool dcurl_init()
 {
-    bool ret = true;
+    bool ret = false;
 
 #if defined(ENABLE_AVX)
-    ret &= registerImplContext(&PoWAVX_Context);
+    if (registerImplContext(&PoWAVX_Context)) {
+        initStatus = initStatus | AVX_BIT;
+        ret |= true;
+    }
 #elif defined(ENABLE_SSE)
-    ret &= registerImplContext(&PoWSSE_Context);
+    if (registerImplContext(&PoWSSE_Context)) {
+        initStatus = initStatus | SSE_BIT;
+        ret |= true;
+    }
 #elif defined(ENABLE_GENERIC)
-    ret &= registerImplContext(&PoWC_Context);
+    if (registerImplContext(&PoWC_Context)) {
+        initStatus = initStatus | C_BIT;
+        ret |= true;
+    }
 #endif
 
 #if defined(ENABLE_OPENCL)
-    ret &= registerImplContext(&PoWCL_Context);
+    if (registerImplContext(&PoWCL_Context)) {
+        initStatus = initStatus | GPU_BIT;
+        ret |= true;
+    }
 #endif
 
 #if defined(ENABLE_FPGA_ACCEL)
-    ret &= registerImplContext(&PoWFPGAAccel_Context);
+    if (registerImplContext(&PoWFPGAAccel_Context)) {
+        initStatus = initStatus | FPGA_BIT;
+        ret |= true;
+    }
 #endif
 
 #if defined(ENABLE_REMOTE)
-    ret &= initializeRemoteContext(&Remote_Context);
+    if (registerRemoteContext(&Remote_Context)) {
+        initStatus = initStatus | REMOTE_BIT;
```

Comment:
I would prefer the statement:
```
initStatus |= REMOTE_BIT;
```

It would be shorter and meaningful.

Reply:
Yes, I've noticed that.
I'll fix it in the next force push.

Reply:
Done.